proto/cacheclient.proto

```diff
@@ -402,16 +402,99 @@ message _ListRemoveRequest {
 
 message _ListRemoveResponse {}
 
+// List fetch + retain index behavior
+// This follows redis behavior for the most part, differences
+// 1. end_index is exclusive, unless Unbounded
+// 2. `start_index` == 0, then we start at the front of list
+// 2. `end_index` == Unbounded, start at the end of the index, inclusive
+/*
+  list_ = [0, 1, 2, 3, 4, 5, 6, 7, 8, 9]
+
+  # Using the slice notation:
+  # start:end
+
+
+  # start=0, end=Unbounded
+  list_[0:] == [0, 1, 2, 3, 4, 5, 6, 7, 8, 9]
+
+
+  # start=5, end=Unbounded
+  list_[5:] == [5, 6, 7, 8, 9]
+
+  # start=5, end=0
+  list_[5:0] == []
+
+  # start=-5, end=Unbounded
+  list_[-5:] == [5, 6, 7, 8, 9]
+
+  # start=-5, end=0
+  list_[-5:0] == []
+
+
+  # start=0, end=5
+  list_[0:5] == [0, 1, 2, 3]
+
+  # start=Unbounded, end=5
+  list_[0:5] == [0, 1, 2, 3]
+
+  # start=0, end=-5
+  list_[0:-5] == [0, 1, 2, 3, 4]
+
+  # start=Unbounded, end=-5
+  list_[0:-5] == [0, 1, 2, 3, 4]
+
+
+  # start=2, end=6
+  list_[2:6] == [2, 3, 4, 5]
+
+  # start=2, end=-6
+  list_[2:-6] == [2, 3]
+
+  # start=-7, end=-2
+  list_[-7:-2] == [3, 4, 5, 6, 7]
+
+  // if using ltrim, this will remove all elements in the list
+  # start=-2, end=-6
+  list_[2:-6] == []
+
+  # empty list
+  list_[-1:1] == []
+ */
+message Unbounded { }
+
 message _ListFetchRequest {
   bytes list_name = 1;
   // Inclusive. 0 (start of list) by default.
   // A negative index counts from the end of the list
-  sint32 start_index = 2;
-  // If unset or 0, this effectively means list.length()
+  oneof start_index {
+    Unbounded unbounded_start = 2;
+    sint32 inclusive_start = 3;
+  }
```

Comment:
does `0` mean the same thing as `Unbounded`? Should we consider not using Unbounded for start indices?